src/constants/fixed_bases.rs

```diff
@@ -0,0 +1,355 @@
+//! Orchard fixed bases.
+use super::{L_ORCHARD_SCALAR, L_VALUE};
+use crate::circuit::gadget::ecc::{chip::FixedPoint, FixedPoints};
+
+use arrayvec::ArrayVec;
+use ff::Field;
+use group::Curve;
+use halo2::arithmetic::lagrange_interpolate;
+use pasta_curves::{
+    arithmetic::{CurveAffine, FieldExt},
+    pallas,
+};
+
+pub mod commit_ivk_r;
+pub mod note_commit_r;
+pub mod nullifier_k;
+pub mod spend_auth_g;
+pub mod value_commit_r;
+pub mod value_commit_v;
+
+/// SWU hash-to-curve personalization for the spending key base point and
+/// the nullifier base point K^Orchard
+pub const ORCHARD_PERSONALIZATION: &str = "z.cash:Orchard";
+
+/// SWU hash-to-curve personalization for the value commitment generator
+pub const VALUE_COMMITMENT_PERSONALIZATION: &str = "z.cash:Orchard-cv";
+
+/// SWU hash-to-curve value for the value commitment generator
+pub const VALUE_COMMITMENT_V_BYTES: [u8; 1] = *b"v";
+
+/// SWU hash-to-curve value for the value commitment generator
+pub const VALUE_COMMITMENT_R_BYTES: [u8; 1] = *b"r";
+
+/// SWU hash-to-curve personalization for the note commitment generator
+pub const NOTE_COMMITMENT_PERSONALIZATION: &str = "z.cash:Orchard-NoteCommit";
+
+/// SWU hash-to-curve personalization for the IVK commitment generator
+pub const COMMIT_IVK_PERSONALIZATION: &str = "z.cash:Orchard-CommitIvk";
+
+/// Window size for fixed-base scalar multiplication
+pub const FIXED_BASE_WINDOW_SIZE: usize = 3;
+
+/// $2^{`FIXED_BASE_WINDOW_SIZE`}$
+pub const H: usize = 1 << FIXED_BASE_WINDOW_SIZE;
+
+/// Number of windows for a full-width scalar
+pub const NUM_WINDOWS: usize =
+    (L_ORCHARD_SCALAR + FIXED_BASE_WINDOW_SIZE - 1) / FIXED_BASE_WINDOW_SIZE;
+
+/// Number of windows for a short signed scalar
+pub const NUM_WINDOWS_SHORT: usize =
+    (L_VALUE + FIXED_BASE_WINDOW_SIZE - 1) / FIXED_BASE_WINDOW_SIZE;
```

Comment:
I think the chip could work over other constants as well. We currently assume these constants in the chip impl because it's too annoying to provide them all as const generics without `const_evaluatable_checked`.